tests/IntegrationTests/WebAppCallsApiCallsGraphUiTests/TestingWebAppCallsApiCallsGraphLocally.cs

```diff
@@ -0,0 +1,124 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Identity.Lab.Api;
+using Microsoft.Playwright;
+using WebAppUiTests;
+using Xunit;
+using Xunit.Abstractions;
+using Process = System.Diagnostics.Process;
+
+namespace WebAppCallsApiCallsGraphUiTests
+{
+#if !FROM_GITHUB_ACTION
+    public class TestingWebAppCallsApiCallsGraphLocally
+    {
+        private const string LocalhostUrl = @"https://localhost:";
+        private const string DevAppPath = @"DevApps\WebAppCallsWebApiCallsGraph";
+        private const string TodoListServicePath = @"\TodoListService";
+        private const string TodoListClientPath = @"\Client";
+        private const string GrpcPath = @"\gRPC";
+        private const string TodoListServiceExecutable = @"\TodoListService.exe";
+        private const string TodoListClientExecutable = @"\TodoListClient.exe";
+        private const string GrpcExecutable = @"\grpc.exe";
+        private const string TodoListServicePort = "44351";
+        private const string TodoListClientPort = "44321";
+        private const string GrpcPort = "5001";
+        private const string SignOutPagePath = @"/MicrosoftIdentity/Account/SignedOut";
+        private const string TodoTitle1 = "Testing create todo item";
+        private const string TodoTitle2 = "Testing edit todo item";
+        private string UiTestAssemblyLocation = typeof(TestingWebAppCallsApiCallsGraphLocally).Assembly.Location;
+        private readonly ITestOutputHelper _output;
+
+        public TestingWebAppCallsApiCallsGraphLocally(ITestOutputHelper output)
+        {
+            _output = output;
+        }
+
+        [Fact]
+        public async Task ChallengeUser_MicrosoftIdentityFlow_LocalApp_ValidEmailPasswordCreds_TodoAppFunctionsCorrectly()
+        {
+            // Arrange web app setup
+            Environment.SetEnvironmentVariable("ASPNETCORE_ENVIRONMENT", "Development");
+            Process? grpcProcess = UiTestHelpers.StartProcessLocally(UiTestAssemblyLocation, DevAppPath + GrpcPath, GrpcExecutable, GrpcPort);
+            Process? clientProcess = UiTestHelpers.StartProcessLocally(UiTestAssemblyLocation, DevAppPath + TodoListClientPath, TodoListClientExecutable, TodoListClientPort);
+            Process? serviceProcess = UiTestHelpers.StartProcessLocally(UiTestAssemblyLocation, DevAppPath + TodoListServicePath, TodoListServiceExecutable, TodoListServicePort);
+            try
+            {
+                if ( !UiTestHelpers.ProcessesAreAlive(new List<Process>() { clientProcess!, serviceProcess!, grpcProcess! }))
+                    {
+                        Assert.Fail($"Could not run web app locally.");
+                    }
+
+                // Arrange Playwright setup
+                using var playwright = await Playwright.CreateAsync();
+                IBrowser browser;
+                browser = await playwright.Chromium.LaunchAsync(new() { Headless = true }); // To see the browser UI, set Headless = false
```

Comment:
```suggestion
                // To see the browser UI, set Headless = false
                browser = await playwright.Chromium.LaunchAsync(new() { Headless = true }); 
```
I think it's convention to have comments on their own line, not at the end of a line of code